### DISCUSSION TITLE
uninstalling skype

### DISCUSSION THREAD

**Discussion by assiaBenot** [community/community#127303] [Programming Help]
### Body

I am trying to learn PowerShell with PDQ and I want to delete Skype using a PowerShell script via PDQ Connect. I have tried all the solutions provided by ChatGPT, but they didn't work. I tried executing a script on my Windows PowerShell ISE, and it worked. However, I want to find a way to uninstall Skype remotely with a PDQ Connect package.

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mranv** [comment]
To uninstall Skype remotely using PDQ Connect with a PowerShell script, follow these steps:

### Step 1: Create a PowerShell Script
Create a PowerShell script that uninstalls Skype. Save the script as `UninstallSkype.ps1`. Here is an example of a script that uninstalls Skype:

```powershell
# Define Skype product name
$SkypeProductName = "Skype"

# Get the Skype application
$SkypeApp = Get-WmiObject -Class Win32_Product | Where-Object { $_.Name -like "*$SkypeProductName*" }

# Check if Skype is installed
if ($SkypeApp) {
    # Uninstall Skype
    $SkypeApp.Uninstall()
    Write-Output "Skype has been uninstalled."
} else {
    Write-Output "Skype is not installed on this system."
}
```

### Step 2: Test the Script Locally
Ensure that the script works correctly by running it locally on a test machine using Windows PowerShell ISE or the PowerShell console.

### Step 3: Create a PDQ Connect Package
1. **Open PDQ Connect**: Launch the PDQ Connect application on your management computer.
2. **Create a New Package**: Click on "New Package" to create a new deployment package.
3. **Add a PowerShell Step**:
    - **Step Name**: Name the step something like "Uninstall Skype".
    - **Step Type**: Choose "PowerShell".
    - **PowerShell Script**: Copy and paste the content of `UninstallSkype.ps1` into the PowerShell script box.

### Step 4: Configure Package Options
Configure the package options as needed:
- **Target Computers**: Specify the target computers where Skype should be uninstalled.
- **Run As**: Ensure the script runs with appropriate privileges (e.g., as an administrator).

### Step 5: Deploy the Package
1. **Save the Package**: Save your newly created package.
2. **Deploy the Package**: Select the target computers or a group of computers and deploy the package.

### Example Deployment Script in PDQ Connect
For a streamlined process, here’s a refined version of the script that could be directly used in PDQ Connect:

```powershell
# Define Skype product name
$SkypeProductName = "Skype*"

# Get the Skype application
$SkypeApp = Get-WmiObject -Class Win32_Product | Where-Object { $_.Name -like $SkypeProductName }

# Check if Skype is installed
if ($SkypeApp) {
    # Uninstall Skype
    $SkypeApp.ForEach({ $_.Uninstall() })
    Write-Output "Skype has been uninstalled."
} else {
    Write-Output "Skype is not installed on this system."
}
```

### Troubleshooting Tips
- **Permissions**: Ensure the PDQ Connect package runs with sufficient privileges to uninstall applications.
- **Logs**: Check PDQ Connect logs for any error messages if the deployment fails.
- **Network Connectivity**: Ensure that the target machines are accessible and that there are no network issues blocking the deployment.

This approach should allow you to remotely uninstall Skype from multiple computers using PDQ Connect.
